lib/openid_connect.ex

```diff
@@ -126,6 +126,32 @@ defmodule OpenidConnect do
     |> URI.to_string()
   end
 
+
   defp assert_json(%{"error" => reason}), do: {:error, reason}
   defp assert_json(json), do: {:ok, json}
+
+  @spec remaining_lifetime([{String.t, String.t}]) :: integer | nil
+  defp remaining_lifetime(headers) do
+    with headers = Enum.into(headers, %{}),
+         {:ok, max_age} <- find_max_age(headers),
+         {:ok, age} <- find_age(headers) do
+      max_age - age
+    else
+      _ -> nil
+    end
+  end
+
+  defp find_max_age(headers) when is_map(headers) do
+    case Regex.run(~r"(?<=max-age=)\d+", Map.get(headers, "Cache-Control", "")) do
```

Comment:
How confident are we in using regex here? Is there an existing work we can crib from?

Reply:
The naive way would split the string and match it like `"max-age=" <> max_age = header`. I'm confident with the regex.